src/iottalk/DAI.java

```diff
@@ -152,9 +155,19 @@ private boolean _onSignal(String command, String df){
                 return true;  //already connect
             }
             DeviceFeature dft = dfMap.get(df);
+            if (dft == null){
```

Comment:
這樣就好了吧？ dft 之後應該都用不到，就不要這個沒意義的命名了
```suggestion
            if (dft == dfMap.get(df)){ 
```